pkg/deploy/lattice/listener_manager.go

```diff
@@ -70,16 +72,32 @@ func (d *defaultListenerManager) Upsert(
 	}
 
 	// no listener currently exists, create
-	defaultStatus := aws.Int64(404)
-	defaultResp := vpclattice.FixedResponseAction{
-		StatusCode: defaultStatus,
+	defaultAction := vpclattice.RuleAction{
+		FixedResponse: &vpclattice.FixedResponseAction{
+			StatusCode: aws.Int64(404),
+		},
 	}
 
+	if modelListener.Spec.Protocol == vpclattice.ListenerProtocolTlsPassthrough {
+		// Fill the defaultAction tgs for TLS_PASSTHROUGH lattice listener
+		var latticeTGs []*vpclattice.WeightedTargetGroup
+		for _, modelTG := range defaultActionTgs {
+			latticeTG := vpclattice.WeightedTargetGroup{
+				TargetGroupIdentifier: aws.String(modelTG.LatticeTgId),
+				Weight:                aws.Int64(modelTG.Weight),
+			}
+			latticeTGs = append(latticeTGs, &latticeTG)
+		}
+		d.log.Debugf("For TLS_PASSTHROUGH listener, forward to default target groups %v", latticeTGs)
+		defaultAction = vpclattice.RuleAction{
+			Forward: &vpclattice.ForwardAction{
+				TargetGroups: latticeTGs,
```

Comment:
Plan to do a refactoring for this part, for example:


```
func (l *listenerSynthesizer) getLatticeServiceListenerDefaultAction(ctx context.Context, modelListenerProtocol string) (
	*vpclattice.RuleAction, error,
) {
	if modelListenerProtocol != vpclattice.ListenerProtocolTlsPassthrough {
		return &vpclattice.RuleAction{
			FixedResponse: &vpclattice.FixedResponseAction{
				StatusCode: aws.Int64(404),
			},
		}, nil
	}

	var defaultActionTgs []*model.RuleTargetGroup
	var stackRules []*model.Rule
	_ = l.stack.ListResources(&stackRules)
	if err := l.tgManager.ResolveRuleTgIds(ctx, stackRules[0], l.stack); err != nil {
		return nil, fmt.Errorf("failed to resolve rule tg ids, err = %v", err)
	}
	// Fill the default action target groups for TLS_PASSTHROUGH listener, since TLS_PASSTHROUGH listener only has the defaultAction and no extra listener rules
	defaultActionTgs = stackRules[0].Spec.Action.TargetGroups

	// Fill the defaultAction tgs for TLS_PASSTHROUGH lattice listener
	var latticeTGs []*vpclattice.WeightedTargetGroup
	for _, modelTG := range defaultActionTgs {
		latticeTG := vpclattice.WeightedTargetGroup{
			TargetGroupIdentifier: aws.String(modelTG.LatticeTgId),
			Weight:                aws.Int64(modelTG.Weight),
		}
		latticeTGs = append(latticeTGs, &latticeTG)
	}

	l.log.Debugf("For TLS_PASSTHROUGH listener, forward to default target groups %v", latticeTGs)
	return &vpclattice.RuleAction{
		Forward: &vpclattice.ForwardAction{
			TargetGroups: latticeTGs,
		},
	}, nil

}

```

and don't expose logic about TLS_PASSTHROUGH in the `func (d *defaultListenerManager) Upsert(
	ctx context.Context,
	modelListener *model.Listener,
	modelSvc *model.Service,
	defaultAction *vpclattice.RuleAction) ` 

and  `func (l *listenerSynthesizer) Synthesize(ctx context.Context) `